KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift

```diff
@@ -181,6 +180,88 @@ final class KarhooNewCheckoutViewModel: ObservableObject {
         }
     }
 
+    // MARK: Interactions
+
+    func didTapPassenger() {
+        // TODO: - handle passenger flow
+    }
+
+    func didTapOptions() {
+        // TODO: - handle options flow
+    }
+
+    func didTapFlightNumber() {
+        // TODO: - handle flight number flow
+    }
+
+    func didSetTermsAndConditions(_ termsAndConditionsSelected: Bool) {
+        // TODO: - handle t&c flow
+    }
+
+    func didSetComment(_ comment: String) {
+        // TODO: - handle comment flow
+    }
+
+    func didTapConfirm() {
+        // MARK: - Validate & proceed with payment flow
+        guard validateIfAllRequiredDataAreProvided() else {
+            return
+        }
+        submitBooking()
+    }
+
+    // MARK: - Booking
+
+    private func submitBooking() {
+        paymentsWorker.performBooking()
+    }
+
+    // MARK: - Analytics
+
+    private func reportScreenOpened() {
+        analytics.checkoutOpened(quote)
+    }
+
+    private func reportBookingEvent(quoteId: String) {
+        analytics.bookingRequested(quoteId: quoteId)
+    }
+
+    private func reportBookingSuccess(tripId: String, quoteId: String?, correlationId: String?) {
+        analytics.bookingSuccess(tripId: tripId, quoteId: quoteId, correlationId: correlationId)
+    }
+
+    private func reportBookingFailure(message: String, correlationId: String?) {
+        analytics.bookingFailure(
+            quoteId: quote.id,
+            correlationId: correlationId ?? "",
+            message: message,
+            lastFourDigits: paymentsWorker.getPaymentNonce()?.lastFour ?? "",
+            paymentMethodUsed: String(describing: KarhooUISDKConfigurationProvider.configuration.paymentManager),
+            date: Date(),
+            amount: quote.price.highPrice,
+            currency: quote.price.currencyCode
+        )
+    }
+
+    private func reportCardAuthorisationSuccess() {
+        analytics.cardAuthorisationSuccess(quoteId: quote.id)
+    }
+
+    private func reportCardAuthorisationFailure(message: String) {
```

Comment:
Shouldn't the payment worker send the payment analytics events be it success or failure?

Reply:
It's a draft. If payment worker do have data for it (quote, for instance), we can do it this way).